app/api/v3/token.py

```diff
@@ -0,0 +1,1775 @@
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+
+See the License for the specific language governing permissions and
+limitations under the License.
+
+SPDX-License-Identifier: Apache-2.0
+"""
+from cerberus import Validator
+from eth_utils import to_checksum_address
+from falcon import Request
+from sqlalchemy import (
+    desc,
+)
+from typing import Optional
+
+from app import log
+from app.api.common import BaseResource
+from app.errors import (
+    InvalidParameterError,
+    NotSupportedError
+)
+from app import config
+from app.model.blockchain import (
+    BondToken,
+    ShareToken,
+    CouponToken,
+    MembershipToken
+)
+from app.utils.web3_utils import Web3Wrapper
+from app.model.db import (
+    Listing,
+    IDXBondToken,
+    IDXShareToken,
+    IDXMembershipToken,
+    IDXCouponToken
+)
+
+LOG = log.get_logger()
+
+
+class StraightBondTokens(BaseResource):
+    """
+    Endpoint: /v3/Token/StraightBond
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.web3 = Web3Wrapper()
+
+    def on_get(self, req, res, **kwargs):
+        session = req.context["session"]
+
+        if config.BOND_TOKEN_ENABLED is False:
+            raise NotSupportedError(method='GET', url=req.path)
+
+        # Validation
+        request_json = StraightBondTokens.validate(req)
+
+        target_address_list = [to_checksum_address(address) for address in request_json["address_list"]]
+        owner_address: Optional[str] = request_json.get("owner_address", None)
+        name: Optional[str] = request_json.get("name", None)
+        symbol: Optional[str] = request_json.get("symbol", None)
+        company_name: Optional[str] = request_json.get("company_name", None)
+        tradable_exchange: Optional[str] = request_json.get("tradable_exchange", None)
+        status: Optional[bool] = request_json.get("status", None)
+        personal_info_address: Optional[str] = request_json.get("personal_info_address")
+        transferable: Optional[bool] = request_json.get("transferable", None)
+        is_offering: Optional[bool] = request_json.get("is_offering", None)
+        transfer_approval_required: Optional[bool] = request_json.get("transfer_approval_required", None)
+        is_redeemed: Optional[bool] = request_json.get("is_redeemed", None)
+
+        sort_item = "created" if request_json["sort_item"] is None else request_json["sort_item"]
+        sort_order = 0 if request_json["sort_order"] is None else request_json["sort_order"]  # default: asc
+        offset = request_json["offset"]
+        limit = request_json["limit"]
+
+        # 取扱トークンリストを取得
+        # 公開属性によるフィルタリングを行うためJOIN
+        query = session.query(IDXBondToken).\
+            join(Listing, Listing.token_address == IDXBondToken.token_address).\
+            filter(Listing.is_public == True)
```

Comment:
Cache ttl filtering is not implemented because we expect 2 following functionality in this endpoint.
- offset/limit paging
- token attribute filtering